src/App.jsx

```diff
@@ -14,33 +15,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ABC = '1';
+const L = '2';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function Sort(arr, reverse, sort) {
+  let tempArr = [...arr];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sort) {
+    case ABC:
+      tempArr = tempArr.sort((a, b) => a.localeCompare(b));
+      break;
+    case L:
+      tempArr = tempArr.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    tempArr.reverse();
+  }
+
+  return tempArr;
+}
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sort, setSort] = useState('');
+
+  const visibleGoods = Sort(goodsFromServer, isReverse, sort);
+
+  function Reset() {
+    setIsReverse(false);
+    setSort('');
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-danger ${sort === ABC ? null : 'is-light'}`}
+          onClick={() => setSort(ABC)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-danger ${sort === L ? null : 'is-light'}`}
+          onClick={() => setSort(L)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-danger ${isReverse ? null : 'is-light'}`}
+          onClick={() =>
+            !isReverse ? setIsReverse(true) : setIsReverse(false)
+          }
+        >
+          Reverse
+        </button>
+
+        {isReverse || sort ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => Reset()}
+          >
+            Reset
+          </button>
+        ) : null}
+      </div>
+
+      <ul>
+        {visibleGoods.map(goods => (
+          <li data-cy="Good">{goods}</li>
```

Comment:
The `data-cy` attribute should have a unique value for each list item to facilitate testing. Consider using a unique property of the item or the index to differentiate them.